source/MaterialXGenGlsl/Nodes/LightSamplerNodeGlsl.cpp

```diff
@@ -32,6 +32,7 @@ void LightSamplerNodeGlsl::emitFunctionDefinition(const ShaderNode&, GenContext&
         shadergen.emitLine(SAMPLE_LIGHTS_FUNC_SIGNATURE, stage, false);
         shadergen.emitScopeBegin(stage);
         shadergen.emitLine("result.intensity = vec3(0.0)", stage);
+        shadergen.emitLine("result.direction = vec3(0.0)", stage);
```

Comment:
This didn't compile when cross-compiled to HLSL because all members of an out struct argument must be initialized in HLSL.